src/host.rs

```diff
@@ -388,4 +437,54 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn poll_accept() {
```

Comment:
Could we test this from the `TcpListener` level in the existing `tcp.rs` test? You wouldn't need to mock out anything then.

Reply:
@mcches, I just got back into it, and I'm not quite sure how this would be done. First, the `accept` method basically just calls the `poll_accept` method, so we are basically pretty well covered here. For the specific case we're testing here, we require a `Context`. If we're testing from the higher level `TcpStream`, we can get such a waker from `poll_fn`, but I'm not sure what we can do from there: we can't synchronize the client and the server from within  `poll_fn`, we can't inspect what happens with the waker, etc...

Maybe you see something that I don't?